iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/toggle/IcebergFeatureGate.java

```diff
@@ -0,0 +1,16 @@
+package com.linkedin.openhouse.internal.catalog.toggle;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Used to gate the feature of the corresponding method where this annotation is applied, on the
+ * granularity of Iceberg table identified by {@link org.apache.iceberg.catalog.TableIdentifier}
+ */
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Please add more documentation around the expectations (first parameter being tableidentifier), behaviors (throw exceptions), and an example. Coder typically wouldn't hunt for the Aspect file instead rely on annotation to provide all the information.

Reply:
addressed.